scripts/dts/edtsdatabase.py

```diff
@@ -0,0 +1,193 @@
+
+    ##
+    # @brief Insert property value for the device of the given device id.
+    #
+    # @param device_id
+    # @param property_path Path of the property to access
+    #                      (e.g. 'reg/0', 'interrupts/prio', 'label', ...)
+    # @param property_value value
+    #
+    def insert_device_property(self, device_id, property_path, property_value):
+        # special properties
+        if property_path == 'compatible':
+            self._update_device_compatible(device_id, property_value)
+        elif property_path == 'device-type':
+            self._update_device_type(device_id, property_value)
+
+        # normal property management
+        if device_id not in self._edts['devices']:
+            self._edts['devices'][device_id] = dict()
+            self._edts['devices'][device_id]['device-id'] = device_id
+        if property_path == 'device-id':
+            # already set
+            return
+        keys = property_path.strip("'").split('/')
+        property_access_point = self._edts['devices'][device_id]
+        for i in range(0, len(keys)):
+            if i < len(keys) - 1:
+                # there are remaining keys
+                if keys[i] not in property_access_point:
+                    property_access_point[keys[i]] = dict()
+                property_access_point = property_access_point[key[i]]
+            else:
+                # we have to set the property value
+                if keys[i] in property_access_point:
+                    # There is already a value set
+                    current_value = property_access_point[keys[i]]
+                    if not isinstance(current_value, list):
+                        current_value = [current_value, ]
+                    if isinstance(property_value, list):
+                        property_value = current_value.extend(property_value)
+                    else:
+                        property_value = current_value.append(property_value)
+                property_access_point[keys[i]] = property_value
+
+    ##
+    #
+    # @return list of devices that are compatible
+    def compatible_devices(self, compatible):
+        devices = list()
+        for device_id in self._edts['compatibles'][compatible]:
+            devices.append(self._edts['devices'][device_id])
+        return devices
+
+    ##
+    #
+    # @return list of device ids of activated devices that are compatible
+    def compatible_devices_id(self, compatible):
+        return self._edts['compatibles'].get(compatible, [])
+
+    ##
+    # @brief Get device tree property value for the device of the given device id.
+    #
+    #
+    # @param device_id
+    # @param property_path Path of the property to access
+    #                      (e.g. 'reg/0', 'interrupts/prio', 'device_id', ...)
+    # @return property value
+    #
+    def device_property(self, device_id, property_path, default="<unset>"):
+        property_value = self._edts['devices'][device_id]
```

Comment:
To be updated vs the latest version.
(And will likely require a more efficient (and iterative) parsing function)